src/prettytable/prettytable.py

```diff
@@ -1887,7 +1887,10 @@ def _stringify_title(self, title, options):
         )
         bits.append(endpoint)
         title = " " * lpad + title + " " * rpad
-        bits.append(self._justify(title, len(self._hrule) - 2, "c"))
+        lpad, rpad = self._get_padding_widths(options)
+        total_widths = [n + lpad + rpad + 1 for n in self._widths]
+
+        bits.append(self._justify(title, sum(total_widths) - 1, "c"))
```

Comment:
Creating and summing a tuple is nearly 2x as fast as a list:

```suggestion
        sum_widths = sum((n + lpad + rpad + 1 for n in self._widths))

        bits.append(self._justify(title, sum_widths - 1, "c"))
```